ref-exchange/src/owner.rs

```diff
@@ -18,6 +19,33 @@ impl Contract {
         self.owner_id.clone()
     }
 
+    /// Retrieve NEP-141 tokens that not mananged by contract to owner,
+    /// Caution: Must check that `amount <= total_amount_in_account - amount_managed_by_contract` before calling !!!
+    /// Returns promise of ft_transfer action.
+    #[payable]
+    pub fn retrieve_unmanaged_token(&mut self, token_id: ValidAccountId, amount: U128) -> Promise {
+        self.assert_owner();
+        assert_one_yocto();
+        let token_id: AccountId = token_id.into();
+        let amount: u128 = amount.into();
+        assert!(amount > 0, "{}", ERR29_ILLEGAL_WITHDRAW_AMOUNT);
```

Comment:
should `amount <= total_amount_in_account - amount_managed_by_contract` be asserted here?

Reply:
From contract side, It is hard to tell cause managed_amount is spreaded all over the state in the form of LookupMap. As an owner interface, we can reqeust a pre-call check by the professional caller. Furthermore, it is always used in this scenario, that someone make a mistake to simply transfer some assets to the contract account ID. In that case, we  can get the refund amount from the transfer TX safely.